eng/tools/ChangeValidation/ExclusionFileValidation.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+
+using static ValidateVmrChanges.Validate;
+
+namespace ValidateVmrChanges;
+
+internal class ExclusionFileValidation
+{
+    private static readonly string SourceMappingsPath = "src/source-mappings.json";
+
+    internal static List<ProcessingMessage> VerifyFileExclusions(List<string> diffFiles, string targetBranch)
+    {
+        List<ProcessingMessage> messages = [];
+
+        string remoteTargetBranch = "origin/" + targetBranch;
+        var originalExclusionRules = GetExclusionPatternsFromBranch(remoteTargetBranch);
+
+        if (originalExclusionRules == null || !originalExclusionRules.Any())
+        {
+            AddProcessingMessage(messages, Error($"No exclusion rules found in `{SourceMappingsPath}` on the target branch, or the file does not exist."));
+            return messages;
+        }
+
+        var newExclusionRules = GetExclusionPatternsFromBranch("HEAD");
+
+        if (newExclusionRules == null || !newExclusionRules.Any())
```

Comment:
It's entirely possible we will not have any exclusions and that is a valid scenario